### PR TITLE
Migrate table facade

### PR DESCRIPTION
**Pull request by rpiaggio**

### CODE REVIEW

File: common-graphql/src/main/scala/queries/schema/UserPreferencesDB.scala [hunk 1]
```diff
@@ -24,4 +26,10 @@ trait UserPreferencesDB:
   object Enums:
     type ItcChartType            = enums.ItcChartType
     type LucumaTableIdsEnum      = explore.model.enums.TableId
-    type LucumaSortDirectionEnum = explore.model.enums.SortDirection
+    type LucumaSortDirectionEnum = SortDirection
+
+  given Enumerated[SortDirection] =
```
Comment (cquiroz) [inline common-graphql/src/main/scala/queries/schema/UserPreferencesDB.scala:31]: Why not to define the enumerated where `SortDirection` is defined?

File: explore/src/main/scala/explore/config/SequenceTable.scala [hunk 2]
```diff
@@ -109,57 +110,82 @@ object SequenceTable:
...
     ColDef(
-      "q",
+      QColumnId,
       _.q,
       header = _ => rightAligned("q"),
       cell = cell => rightAligned(offsetFormat.format(cell.value))
     ),
     ColDef(
-      "lambda",
+      LambdaColumnId,
```
Comment (cquiroz) [inline explore/src/main/scala/explore/config/SequenceTable.scala:162]: Nice I prefer this to plain strings

File: explore/src/main/scala/explore/targeteditor/BrightnessesEditor.scala [hunk 3]
```diff
@@ -138,9 +144,9 @@ sealed abstract class BrightnessesEditorBuilder[T, Props <: BrightnessesEditor[T
                       onClick = brightnesses.mod(_ - cell.value)
                     )(Icons.Trash)
                   ),
-                size = 20,
-                minSize = 20,
-                maxSize = 20,
+                size = 20.toPx,
+                minSize = 20.toPx,
+                maxSize = 20.toPx,
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/BrightnessesEditor.scala:149]: scalajs react has a `.px` syntax, could we reuse it?

File: explore/src/main/scala/explore/targeteditor/TargetTable.scala [hunk 4]
```diff
@@ -91,12 +95,12 @@ object TargetTable extends TableHooks:
       .useMemoBy((props, _) => (props.obsIds, props.targets.get)) { (props, ctx) => _ =>
         import ctx.given
 
-        def column[V](id: String, accessor: SiderealTargetWithId => V) =
-          ColDef(id, accessor, columnNames(id))
+        // def column[V](id: DeleteColumnId, accessor: SiderealTargetWithId => V) =
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/TargetTable.scala:98]: Commented code

**Review by cquiroz — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala [hunk 5]
```diff
@@ -106,7 +124,7 @@ object ConstraintsSummaryTable extends TableHooks:
               ctx.pageUrl(AppTab.Constraints, props.programId, Focused.singleObs(obsId))
 
             List(
-              column("edit", ConstraintGroup.obsIds.get)
+              column(ColumnId("edit"), ConstraintGroup.obsIds.get)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala:127]: Can't you use `EditColumnId` that is defined above?

**Review by toddburnside — APPROVED** [review]
LGTM

File: common-graphql/src/main/scala/queries/schema/UserPreferencesDB.scala (hunk 1 above)
Comment (rpiaggio) [reply common-graphql/src/main/scala/queries/schema/UserPreferencesDB.scala:31]: `SortDirection` is defined in the facade now, which doesn't have `lucuma-core` as dependency.

File: explore/src/main/scala/explore/targeteditor/BrightnessesEditor.scala (hunk 3 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/targeteditor/BrightnessesEditor.scala:149]: `.px` returns a `String`, whereas here we need the numeric value.

Actually, I named this `px` initially, but it clashed with `scalajs-react`'s.

File: explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala (hunk 5 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala:127]: Yup, missed this, thank you.

**Pull request merged**